File: STMS-Server/src/main/java/Data/Food.java
package Data;

import Strategy.IPriceStrategy;
import Strategy.OrderPriceStrategyFactory;

public class Food {
    public String id;
    public String foodClass;    //菜品类型
    public String st;           //IPriceStrategy
    public String name;
    public int price;
    public String foodTip;
    public int foodNum;

    public Food() {
    }

    public Food(String id, String foodClass, String st, String name, int price, String foodTip) {
        this.id = id;
        this.foodClass = foodClass;
        this.st = st;
        this.name = name;
        this.price = price;
        this.foodTip = foodTip;
    }

    /**
     * 计算菜价
     *
     * @return 未初始化返回-1，价格策略错误-2，否则返回该菜品价格
     */
    public int CalculatePrice() {
        try {
            if (id == "" || foodNum < 0)
                return -1;

            IPriceStrategy strategy = OrderPriceStrategyFactory.getStrategy(st);
            if (strategy == null) {
                return -2;
            }
            return strategy.quote(this);
        } catch (Exception e) {
            e.printStackTrace();
            System.out.println("价格计算出错！");
            return -1;
        }
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getFoodClass() {
        return foodClass;
    }

    public void setFoodClass(String foodClass) {
        this.foodClass = foodClass;
    }

    public String getSt() {
        return st;
    }

    public void setSt(String st) {
        this.st = st;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public String getFoodTip() {
        return foodTip;
    }

    public void setFoodTip(String foodTip) {
        this.foodTip = foodTip;
    }

    public void setFoodNum(int foodNum) {
        this.foodNum = foodNum;
    }

    public int getFoodNum() {
        return foodNum;
    }
}

File: DGL/Util.Controls.WPFTest/Food.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Util.Controls.WPFTest
{
    class Food
    {
        public string id { get; set; }
        public string foodClass { get; set; }    //菜品类型
        public string st { get; set; }           //IPriceStrategy 只能是single（单点） 和 weight (称重)
        public string name { get; set; }
        public int price { get; set; }
        public string foodTip { get; set; }
        public int foodNum { get; set; }
        public Food() { }

        public Food(string id, string foodClass, string st, string name, int price, string foodTip)
        {
            this.id = id;
            this.foodClass = foodClass;
            this.st = st;
            this.name = name;
            this.price = price;
            this.foodTip = foodTip;
        }

        public string GetId()
        {
            return id;
        }

        public void SetId(string id)
        {
            this.id = id;
        }

        public string GetFoodClass()
        {
            return foodClass;
        }

        public void SetFoodClass(string foodClass)
        {
            this.foodClass = foodClass;
        }

        public string GetSt()
        {
            return st;
        }

        public void SetSt(string st)
        {
            this.st = st;
        }

        public string GetName()
        {
            return name;
        }

        public void SetName(string name)
        {
            this.name = name;
        }

        public int GetPrice()
        {
            return price;
        }

        public void SetPrice(int price)
        {
            this.price = price;
        }

        public string GetFoodTip()
        {
            return foodTip;
        }

        public void SetFoodTip(string foodTip)
        {
            this.foodTip = foodTip;
        }

        public void SetFoodNum(int foodNum)
        {
            this.foodNum = foodNum;
        }

        public int GetFoodNum()
        {
            return foodNum;
        }
    }
}
